### PR TITLE
[DOC] Add REST API doc for POST /batches with uploading resource

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- add docs in REST API for POST /batches with uploading resource, which was introduced in #4106 .

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request
 
image

### CODE REVIEW

**Comment by bowenliang123** [comment]
This PR is adding REST API docs for resource file uploading in batch creation.
I'm using `POST /batches (with uploading resource)` for the title to distinguish the similar API in same path `POST /batches` accepting different media types. Do you think I should keep this or just use the same title as `POST /batches`  here?
cc @turboFei @pan3793

**Review by pan3793 — APPROVED** [review]

File: docs/client/rest/rest_api.md [hunk 1]
```diff
@@ -371,6 +373,23 @@ Create a new batch.
...
+Create a new batch with uploading resource file.
+
+#### Request Body
+- Media type: `multipart-formdata`
+- Request body structure in multiparts:
+
+| Name         | Description                                                                                      | Media Type       |
+|:-------------|:-------------------------------------------------------------------------------------------------|:-----------------|
+| batchRequest | The batch request in JSON format as request body requried in [POST /batches](#post-batches)      | application/json |
+| resourceFile | The uploading resource to execute, which will be cached on server and cleaned up after execution | File             |
```
Comment (turboFei) [inline docs/client/rest/rest_api.md:387]: The resource to upload and execute?

Comment (turboFei) [reply docs/client/rest/rest_api.md:387]: could you provide the `curl command` for this?

Comment (bowenliang123) [reply docs/client/rest/rest_api.md]: Good idea and the curl example is added to the API description.

**Review by ulysses-you — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Comment by madeirak** [comment]
@bowenliang123 
hi, I encountered some issues while using this feature, like below , how can I obtain the correct response, thx!   :)


image
my usage is like the following, using Python
```
headers = {"Content-Type": "application/json"}
data = {
    'batchRequest': param  # param conforms to the JSON structure and contains data of batchRequest
}

files = {
    "resourceFile": open(script_path, 'r')
}

response = requests.post(
    url, headers=headers, data=data, files=files
)
```

**Comment by bowenliang123** [comment]
Please read and follow the docs in https://kyuubi.readthedocs.io/en/master/client/rest/rest_api.html#post-batches-with-uploading-resource. The required content type of the overall request is `multipart-formdata` (in order to support file resource uploading).

**Comment by madeirak** [comment]
> Please read and follow the docs in https://kyuubi.readthedocs.io/en/master/client/rest/rest_api.html#post-batches-with-uploading-resource. The required content type of the overall request is `multipart-formdata` (in order to support file resource uploading).

when i change the  headers = {"Content-Type": "application/json"} to  headers = {"Content-Type": "multipart/form-data"} meet another problem

HTTP 400 Bad Request   :(

**Comment by bowenliang123** [comment]
Not just the header itself , but make the whole request to conform the required content type.
And you can refer to the example in https://github.com/apache/kyuubi/blob/master/kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/BatchesResourceSuite.scala#L263 and use the REST SDK for Java.

**Pull request closed without merging**